interfaces/sourcegen/src/sourcegen/headers/ct_auto.yaml

```diff
@@ -44,5 +40,40 @@ recipes:
   implements: permeability_0
 - name: epsilon0
   implements: epsilon_0
+- name: getCanteraError
+  brief: Get %Cantera error.
+  what: function
+  declaration: int32_t ct3_getCanteraError(int32_t bufLen, char* buf)
+  parameters:
+    bufLen: Length of reserved array.
+    buf: String containing %Cantera error.
+  returns: Actual length of string or -1 for exception handling.
+  code: |-
+    string err = Application::Instance()->lastErrorMessage();
+    copyString(err, buf, bufLen);
+    return int32_t(err.size());
+- name: setLogWriter
+  brief: Set log writer.
+  what: function
+  declaration: int32_t ct3_setLogWriter(void* logger)
+  parameters:
+    logger: Logger.
+  code: |-
+    Logger* logwriter = (Logger*)logger;
+    setLogger(logwriter);
+    return 0;
```

Comment:
I think we can drop `setLogWriter` from CLib. There's really no way to use it as a pure C interop, since the `Logger` type is a C++ class. And if you're in an interface that can interact with C++ objects, you may as well just call the C++ method directly, as we do in the Cython interface when we specify its custom logger. I suppose it's worth checking how the new MATLAB interface works, though.